srcts/src/shiny/bind.ts

```diff
@@ -12,6 +12,19 @@ import { shinyAppBindOutput, shinyAppUnbindOutput } from "./initedMethods";
 import { sendImageSizeFns } from "./sendImageSize";
 
 type BindScope = HTMLElement | JQuery<HTMLElement>;
+type BindAllScope = HTMLElement | JQuery<HTMLElement> | Text;
```

Comment:
This will require some downstream changes, but it feels like, if `checkValidity()` is gonna support a `Text` type, then so should `.bindAll()`, etc. 

```suggestion
type BindScope = HTMLElement | JQuery<HTMLElement> | Text;
```

Reply:
It's super weird that text is being passed in at all. It bothers me not knowing where that's coming from.

Reply:
The `el.innerHTML || el.textContent` in this code seems like a likely source

https://github.com/rstudio/shiny/blob/20b2669e760d3be9699fc850307e18f088dbdc91/srcts/src/shiny/shinyapp.ts#L1165-L1175

Reply:
Just above it that section I found this comment https://github.com/rstudio/shiny/blob/20b2669e760d3be9699fc850307e18f088dbdc91/srcts/src/shiny/shinyapp.ts#L1145-L1147

Reply:
Anyway, I refactored this to avoid adding a new type, and we just exit validity checking if it's passed something other than an `HTMLElement` or a jQuery instance. 33d66862233b9885657aade531a13db6d1348566

A few take-aways:

1. Validity checks happen with each new `renderContentAsync()` and which nav insertion calls many times in one `nav_insert()`. Thus #4177 will be helpful.
2. There are dragons in the nav insertion logic and I think it's okay to take the easy way out and treat the symptom in `checkValidity()`.

Reply:
> The `el.innerHTML || el.textContent` in this code seems like a likely source

I don't think that's right. It's the type of `el` (i.e., the first argument to `renderContentAsync()`) that's the problem. The type checker incorrectly thinks `el` is always `HTMLElement`, when apparently it can also be `Text`. This can be confirmed by adding `console.log(el)` just before the `renderContentAsync(el, el.innerHTML || el.textContent)` call using the reprex you provided. And in this case, the `Text` nodes are just new lines. Apparently jQuery does this for HTML strings that look like this: `$("<div></div>\n<div></div>")`. Given that, it's really unfortunate that `$divTag`'s type is `JQuery<HTMLElement>` because it really should be something more like `JQuery<HTMLElement | Text>`?

All that said, ideally we'd get to the bottom of the jQuery typing issues, but if that seems like a lot of work/impossible, I think I'd be OK with just adding the `if` check around that `renderContentAsync()` call (along with a comment about how `el`'s type isn't quite right). I can't think of a scenario where `el` is `Text` that you actually want to render?

Reply:
@jcheng5 and I looked through this and the problem is that the `renderContentAsync()` is called on text nodes that are basically just new lines. So I added the check to only call that `renderContentAsync()` for element nodes in 4a01dde46ba86e3ea919bd1a950e5237e529757b.

That said, I remembered while looking at this that about a year ago I ran into issues with a web component in an inserted nav panel being initialized twice and as a result having different behavior when finally added to the DOM because the component changes some attributes on load.

What we'd really like to do is to avoid doing **both** `$(message.divTag.html)` and then later calling `renderContentAsync()`. If you have any advice or context around why we currently do both or how we can get out of the situation, it'd be appreciated!